feat(#967): be opinionated about languages (#972)
medic/cht-docs: content/en/contribute/code/style-guide.md

  Guidelines for writing code
---

## Language

> Our strong preference is to use JavaScript or TypeScript wherever possible, including in webapps, on the server, and for scripting. This is because:

Reviewer: Small wording change. My thinking here is that rather than state "this is how we do it", which someone could read and think, "ok, whatever, that's nice, you have a nice time with that" the suggested wording is more instructive on what we are asking you to do.

```suggestion
Prefer JavaScript or TypeScript wherever possible, including in webapps, on the server, and for scripting. This is because:
```

- every developer on the team knows it already, so we can all maintain it together,
- it makes it easy to write cross-platform code and we have developers on all major operating systems,
- it has a vast number of libraries that are easy to include, and
- it's easy to unit test

> Exceptions to this can be made on a case by case basis, but the decision must be made collectively before coding has begun to avoid having to rewrite.

Reviewer: Just a little grammatical suggestion:

```suggestion
Exceptions to this can be made on a case-by-case basis, but the decision must be made collectively before coding has begun to avoid having to rewrite.
```
